### PR TITLE
PCI calculation

### PR DESCRIPTION
**Pull request by jmhauck**
@MikeTschudi the main thing we are after are the pciUtils functions and a way to easily test them. The components and demo html are purely just that...for demo. We will be taking the output pciUtils.js and calling it from a Survey123 form.

### CODE REVIEW

File: src/components/deduct-calculator/deduct-calculator.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+        type='number'
+      />
+    );
+  }
+
+  /**
+   * Render the distress type input
+   *
+   * @returns a node with a control that shows the distress type name and value
+   * for example ALIGATOR_CRACKING (1)
```
Comment (MikeTschudi) [inline src/components/deduct-calculator/deduct-calculator.tsx:157]: Should this be ALLIGATOR_CRACKING?

File: src/components/deduct-calculator/deduct-calculator.tsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        }
+      </calcite-select>
+    );
+  }
+
+  /**
+   * Render the distress type input
+   *
+   * @returns a node with a control that shows the distress type name and value
+   * for example ALIGATOR_CRACKING (1)
```
Comment (MikeTschudi) [inline src/components/deduct-calculator/deduct-calculator.tsx:175]: ditto

File: src/demos/deduct-calculator.html [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    <link
+      rel="stylesheet"
+      href="https://js.arcgis.com/4.23/esri/themes/light/main.css"
+    />
+    <link rel="stylesheet" href="https://unpkg.com/@esri/calcite-components@1.0.0-beta.94/dist/calcite/calcite.css" type="text/css" />
+    <link rel="stylesheet" href="https://webapps-cdn.esri.com/CDN/fonts/v1.4.1/fonts.css" />
+    <link rel="stylesheet" href="../solutions-components.css" type="text/css">
+
+    <script src="https://js.arcgis.com/4.23/"></script>
```
Comment (MikeTschudi) [inline src/demos/deduct-calculator.html:49]: Current JSAPI is 4.25--should we be using that?

File: src/utils/pciUtils.ts [hunk 4]
```diff
@@ -0,0 +1,780 @@
...
+ * @returns the max CDV value
+ */
+function _getMaxCDV(
+  deductValues: number[],
+  showDebugging: boolean
+): number {
+  let maxCDV;
+  // If none or only one individual deduct value is greater
+  // than two, the total value is used in place of the maximum CD
+  const valuesGraterThan2 = deductValues.filter(deductValue => deductValue > 2);
```
Comment (MikeTschudi) [inline src/utils/pciUtils.ts:615]: let's go with valuesGreaterThan2

File: src/utils/pciUtils.ts [hunk 5]
```diff
@@ -0,0 +1,780 @@
...
+ * @param vals the list of deduct values
+ * @param showDebugging optionally show debugging messages
+ *
+ * @returns the calculated CDV
+ */
+function _getCDV(
+  vals: number[],
+  showDebugging: boolean
+): number {
+  // 9.5.5.1
```
Comment (MikeTschudi) [inline src/utils/pciUtils.ts:735]: What do these comments ("9.5.5.1", "9.5.5.2", "9.5.5.3") mean, please?

**Review by MikeTschudi — APPROVED** [review]
Cool project!  Some fine-tuning and linting, please.

File: src/components/deduct-calculator/deduct-calculator.tsx (hunk 1 above)
Comment (jmhauck) [reply src/components/deduct-calculator/deduct-calculator.tsx:157]: my spellering prowess in action

File: src/demos/deduct-calculator.html (hunk 3 above)
Comment (jmhauck) [reply src/demos/deduct-calculator.html:49]: these actually don't need our JSAPI for anything so I've pulled this

File: src/components/deduct-calculator/deduct-calculator.tsx (hunk 1 above)
Comment (MikeTschudi) [reply src/components/deduct-calculator/deduct-calculator.tsx:157]: "Spellering" is such a fine word!

File: src/utils/pciUtils.ts (hunk 5 above)
Comment (jmhauck) [reply src/utils/pciUtils.ts:729]: These are references to where this portion of the workflow is documented in the ASTM documentation this is based off of.

Comment (MikeTschudi) [reply src/utils/pciUtils.ts:729]: Perhaps mention this? You're one of the few on the team who would know this.

**Pull request merged**